src/core/ext/filters/client_channel/xds/xds_client.cc

```diff
@@ -812,12 +812,17 @@ void XdsClient::ChannelState::AdsCallState::SendMessageLocked(
         GRPC_ERROR_REF(state.error), !sent_initial_message_);
     state.subscribed_resources[xds_client()->server_name_]->Start(Ref());
   } else if (type_url == XdsApi::kRdsTypeUrl) {
-    resource_names.insert(xds_client()->lds_result_->route_config_name);
+    std::string name = (xds_client()->lds_result_.has_value()
+                            ? xds_client()->lds_result_->route_config_name
+                            : xds_client()->server_name_);
```

Comment:
This seems wrong to me.  Why would we assume that the RDS resource name is the server name?  That's never going to be true for TD or for Pilot, and while it's not impossible that there could be some third-party xDS server that does happen to use this name, baking this assumption into the code does not make sense.